app/uk/gov/hmrc/tariffclassificationfrontend/models/Permission.scala

```diff
@@ -16,36 +16,232 @@
-    CANCEL_CASE, ADD_NOTE, ADD_ATTACHMENT, KEYWORDS, EDIT_LIABILITY, EDIT_RULING, APPEAL_CASE, EXTENDED_USE, MOVE_CASE_BACK_TO_QUEUE, VIEW_REPORTS, EDIT_SAMPLE)
+  case object VIEW_REPORTS extends GlobalPermission {
+    override def name: String = nameOf(this)
+    override def appliesTo(operator: Operator): Boolean = managersOnly(operator)
+  }
+
+  case object CREATE_CASES extends GlobalPermission {
+    override def name: String = nameOf(this)
+    override def appliesTo(operator: Operator): Boolean = managersOrTeamMembersOnly(operator)
+  }
+
+  case object ASSIGN_CASE extends GlobalPermission {
+    override def name: String = nameOf(this)
+    override def appliesTo(operator: Operator): Boolean = managersOrTeamMembersOnly(operator)
+  }
+
+  case object RELEASE_CASE extends CasePermission {
+    override def name: String = nameOf(this)
+    override def appliesTo(`case`: Case, operator: Operator): Boolean = managersOrTeamMembersOnly(operator)
+  }
+
+  case object SUPPRESS_CASE extends CasePermission {
+    override def name: String = nameOf(this)
+    override def appliesTo(`case`: Case, operator: Operator): Boolean = managersOrTeamMembersOnly(operator)
+  }
 
-  private val teamBasicPermissions: Set[Permission] = Set(CREATE_CASES, VIEW_MY_CASES, VIEW_QUEUE_CASES, SEARCH, ADVANCED_SEARCH, VIEW_CASES, ASSIGN_CASE, VIEW_CASE_ASSIGNEE,
-    RELEASE_CASE, SUPPRESS_CASE, REOPEN_CASE, CANCEL_CASE, ADD_NOTE, ADD_ATTACHMENT, APPEAL_CASE, EXTENDED_USE, EDIT_SAMPLE)
+  case object REFER_CASE extends CasePermission {
+    override def name: String = nameOf(this)
+    override def appliesTo(`case`: Case, operator: Operator): Boolean = managersOrAssignedTeamMembersOnly(`case`, operator)
+  }
+
+  case object REOPEN_CASE extends CasePermission {
+    override def name: String = nameOf(this)
+    override def appliesTo(`case`: Case, operator: Operator): Boolean = managersOrTeamMembersOnly(operator)
+  }
+
+  case object REJECT_CASE extends CasePermission {
+    override def name: String = nameOf(this)
+    override def appliesTo(`case`: Case, operator: Operator): Boolean = managersOrAssignedTeamMembersOnly(`case`, operator)
+  }
+
+  case object SUSPEND_CASE extends CasePermission {
+    override def name: String = nameOf(this)
+    override def appliesTo(`case`: Case, operator: Operator): Boolean = managersOrAssignedTeamMembersOnly(`case`, operator)
+  }
+
+  case object COMPLETE_CASE extends CasePermission {
+    override def name: String = nameOf(this)
+    override def appliesTo(`case`: Case, operator: Operator): Boolean = managersOrAssignedTeamMembersOnly(`case`, operator)
+  }
+
+  case object REMOVE_ATTACHMENTS extends CasePermission {
+    override def name: String = nameOf(this)
+    override def appliesTo(`case`: Case, operator: Operator): Boolean = managersOrAssignedTeamMembersOnly(`case`, operator)
+  }
+
+  case object CANCEL_CASE extends CasePermission {
```

Comment:
Should this not extend GlobalPermission?  Same with add-not, add-attachment, etc

Reply:
My thoughts were that `GlobalPermission`s  are for when you arent dealing with a single case e.g. Search, Queues etc

Reply:
This will also end up with a status check (`COMPLETED`) when I start pulling that out

Reply:
... in which case you need the case (see what I did there).  Fair enough